### PR TITLE
Add endpiont to get books by status

### LINKED ISSUE #409: Export a list of barcodes in a specific state
From the Google Books app, when you select a state in the left hand column, the user should be able to export a complete list of all of the barcodes in that state and download as a plain text file.

### PR DESCRIPTION
**Pull request by aaron-collier**
## Why was this change made?

Fixes #409 by adding a report/state/:status path with configurable output fields.

## Was the documentation (README, DevOpsDocs, wiki) updated?

README

### CODE REVIEW

File: app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -4,20 +4,50 @@
...
     return nil if books.empty?
 
     CSV.generate(headers: true) do |csv|
       books.each do |barcode|
+        output = []
         book = Book.find(barcode)
-        csv << [barcode, book.public_domain?]
+        output << [barcode, book.public_domain?, book.status, book.scanned_date]
+        output << book.read_attribute(reason_field) unless reason_field.nil?
+        csv << output.flatten
```
Comment (jcoyne) [inline app/controllers/reports_controller.rb]: why do we need to flatten?

Comment (aaron-collier) [reply app/controllers/reports_controller.rb]: The second append here causes the output when put into csv to output the array brackets without the `flatten`:

with flatten:
```
sul.amcollie@li-dl-7121-6cf6 ../DLSS/Infrastructure/google-books amcollie_409 ❯ curl http://localhost:3000/reports/status/DOWNLOAD_FAILED
36105017322285,false,DOWNLOAD_FAILED,2009-02-28 20:54:42 UTC,Conversion failed
```

Without flatten:
```
sul.amcollie@li-dl-7121-6cf6 ../DLSS/Infrastructure/google-books amcollie_409 ❯ curl http://localhost:3000/reports/status/DOWNLOAD_FAILED
"[""36105017322285"", false, ""DOWNLOAD_FAILED"", Sat, 28 Feb 2009 20:54:42 UTC +00:00]",Conversion failed
```

Comment (jcoyne) [reply app/controllers/reports_controller.rb]: @aaron-collier can you change line 48 to be:
```
output << barcode << book.public_domain? << book.status << ...
```

then we don't need that?

Comment (jcoyne) [reply app/controllers/reports_controller.rb]: or `output = [barcode, book.public_domain?, book.status, ...]`

Comment (aaron-collier) [reply app/controllers/reports_controller.rb]: @jcoyne updated. I think that addresses your concern. Thanks!

**Pull request merged**